src/panels/ConnectAcrToClusterPanel.ts

```diff
@@ -0,0 +1,316 @@
+            clusterKey,
+            hasAcrPull,
+        });
+    }
+
+    private async handleCreateAcrRoleAssignmentRequest(
+        acrKey: AcrKey,
+        clusterKey: ClusterKey,
+        webview: MessageSink<ToWebViewMsgDef>,
+    ) {
+        const principalId = await getClusterPrincipalId(this.sessionProvider, clusterKey);
+        if (failed(principalId)) {
+            window.showErrorMessage(getErrorMessage(principalId));
+            return;
+        }
+
+        const client = getAuthorizationManagementClient(this.sessionProvider, acrKey.subscriptionId);
+        const scope = getScopeForAcr(acrKey.subscriptionId, acrKey.resourceGroup, acrKey.acrName);
+
+        const roleAssignment = await createRoleAssignment(
+            client,
+            acrKey.subscriptionId,
+            principalId.result,
+            acrPullRoleDefinitionName,
+            "ServicePrincipal",
+            scope,
+        );
+
+        if (failed(roleAssignment)) {
+            window.showErrorMessage(roleAssignment.error);
+        }
+
+        const roleAssignmentsResult = await getPrincipalRoleAssignmentsForAcr(
+            client,
+            principalId.result,
+            acrKey.resourceGroup,
+            acrKey.acrName,
+        );
+
+        const roleAssignments = defaultAndNotify(roleAssignmentsResult, []);
+        const hasAcrPull = roleAssignments.some(isAcrPull);
+        webview.postCreateAcrRoleAssignmentResponse({
+            acrKey,
+            clusterKey,
+            hasAcrPull,
+        });
+    }
+
+    private async handleDeleteAcrRoleAssignmentRequest(
+        acrKey: AcrKey,
+        clusterKey: ClusterKey,
+        webview: MessageSink<ToWebViewMsgDef>,
+    ) {
+        const principalId = await getClusterPrincipalId(this.sessionProvider, clusterKey);
+        if (failed(principalId)) {
+            window.showErrorMessage(getErrorMessage(principalId));
+            return;
+        }
+
+        const client = getAuthorizationManagementClient(this.sessionProvider, acrKey.subscriptionId);
+        const scope = getScopeForAcr(acrKey.subscriptionId, acrKey.resourceGroup, acrKey.acrName);
+        await deleteRoleAssignment(client, acrKey.subscriptionId, principalId.result, acrPullRoleDefinitionName, scope);
+
+        const roleAssignmentsResult = await getPrincipalRoleAssignmentsForAcr(
+            client,
+            principalId.result,
+            acrKey.resourceGroup,
+            acrKey.acrName,
+        );
+
+        const roleAssignments = defaultAndNotify(roleAssignmentsResult, []);
+        const hasAcrPull = roleAssignments.some(isAcrPull);
+        webview.postDeleteAcrRoleAssignmentResponse({
+            acrKey,
+            clusterKey,
+            hasAcrPull,
+        });
+    }
+}
+
+function isAcrPull(roleAssignment: RoleAssignment): boolean {
+    if (!roleAssignment.roleDefinitionId) {
+        return false;
+    }
+
+    const roleDefinitionName = roleAssignment.roleDefinitionId.split("/").pop();
```

Comment:
can add sample value how roleassingment response would be for future debugging purposes.